### PR TITLE
test: added testing for the exit handlers

### PR DESCRIPTION
**Pull request by 1p22geo**
I know.

It's not perfect.

It's far from perfect.

But this feature should be testable too.

### CODE REVIEW

**Comment by 1p22geo** [comment]
ahhhh the lockfile

**Comment by 1p22geo** [comment]
wdym lockfile missing
it was there at f5c4af3608a41b5d15a123dd5fa5beff168a6a08

the CI at that commit says `ERR_PNPM_NO_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is absent`
https://github.com/shahradelahi/storage-box/actions/runs/10881607810/job/30216477688

but the file is there
https://github.com/shahradelahi/storage-box/blob/f5c4af3608a41b5d15a123dd5fa5beff168a6a08/pnpm-lock.yaml

what is going on
is this PR cursed

**Comment by 1p22geo** [comment]
image
Works on my machine.

**Comment by shahradelahi** [comment]
It's because you're using a different `pnpm` version. Use `corepack enable`; it automatically downloads and uses the package manager specified in `package.json`.

- [Node.js Corepack](https://nodejs.org/api/corepack.html)

**Comment by 1p22geo** [comment]
Once I get home I'll try this. Sounds like it might work. Thanks!

**Comment by 1p22geo** [comment]
image

**Comment by 1p22geo** [comment]
alr how about now

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
   "dependencies": {
     "@msgpack/msgpack": "3.0.0-beta2",
     "debounce": "^2.1.0",
+    "sinon": "^19.0.2",
```
Comment (shahradelahi) [inline package.json]: Testing modules should be installed as devDependencies

**Pull request merged**